osquery/core/query.cpp

```diff
@@ -124,14 +124,15 @@ Status Query::addNewResults(const QueryData& current_qd,
   bool update_db = true;
   if (!fresh_results && calculate_diff) {
     // Get the rows from the last run of this query name.
-    QueryData previous_qd;
+    QueryDataSet previous_qd;
     auto status = getPreviousQueryResults(previous_qd);
     if (!status.ok()) {
       return status;
     }
 
     // Calculate the differential between previous and current query results.
     dr = diff(previous_qd, current_qd);
```

Comment:
'diff()' function implements binary search earlier it was doing liner search. 

Comparison of 'diff based on linear search' Vs 'diff() based on binary search'
Ran query 'select * from processes' 
=======================
With 1000 processes
-------------------------------------------------------------
linear search based diff()  took on avg 122 milli secs
--------------------------------------------------------------
Binary search based diff() took on avg 8 milli secs                                        
-------------------------------------------------------------
============================================
With 10000 processes
--------------------------------------------------------
linear search based diff()  took on avg 22000 milli secs
-----------------------------------------------------------
Binary search based diff() took on avg 97 milli secs
------------------------------------------------------------